Source/scrollrt.cpp

```diff
@@ -2076,150 +2064,159 @@ void __fastcall scrollrt_draw_e_flag(BYTE *pBuff, int x, int y, int a4, int a5,
 
 void __fastcall DrawZoom(int x, int y)
 {
-	int v2;         // edi
-	int v3;         // ebx
-	int v4;         // esi
-	int v5;         // esi
-	int v6;         // edi
-	int v7;         // esi
-	int v8;         // edi
-	int v9;         // esi
-	int v10;        // edi
-	_WORD *v11;     // edi
-	char *v12;      // esi
-	char *v13;      // ebx
-	signed int v14; // edx
-	signed int v15; // ecx
-	short v16;      // ax
-	int v17;        // eax
-	signed int v18; // [esp+Ch] [ebp-10h]
-	signed int v19; // [esp+Ch] [ebp-10h]
-	signed int a5;  // [esp+10h] [ebp-Ch]
-	int a5a;        // [esp+10h] [ebp-Ch]
-	signed int a6;  // [esp+14h] [ebp-8h]
-	signed int a6a; // [esp+14h] [ebp-8h]
-	int a6b;        // [esp+14h] [ebp-8h]
-	int ya;         // [esp+18h] [ebp-4h]
-
-	v2 = ScrollInfo._sxoff + 64;
-	dword_5C2FF8 = 6;
-	dword_5C2FFC = 6;
-	v3 = x - 6;
-	ya = y - 1;
-	a5 = 6;
-	v4 = ScrollInfo._syoff + 143;
+	int i, sx, sy, chunks, blocks;
+	int wdt, nSrcOff, nDstOff;
+
 	scr_pix_width = 384;
 	scr_pix_height = 192;
-	v18 = 3;
-	switch (ScrollInfo._sdir) {
-	case DIR_SW:
-		goto LABEL_3;
-	case DIR_W:
-		a5 = 7;
-	LABEL_3:
-		v4 = ScrollInfo._syoff + 111;
-		v3 = x - 7;
-		ya = y - 2;
-		goto LABEL_9;
-	case DIR_NW:
-		goto LABEL_7;
-	case DIR_N:
-		v18 = 4;
-		goto LABEL_7;
-	case DIR_NE:
-		goto LABEL_9;
-	case DIR_E:
-		v18 = 4;
-		goto LABEL_6;
-	case DIR_SE:
-	LABEL_6:
-		v2 = ScrollInfo._sxoff;
-		v3 = x - 7;
-		ya = y;
-	LABEL_7:
-		a5 = 7;
+	dword_5C2FF8 = 6;
+	dword_5C2FFC = 6;
+
+	sx = ScrollInfo._sxoff + 64;
+	sy = ScrollInfo._syoff + 143;
+	x -= 6;
+	y--;
+	chunks = 6;
+	blocks = 3;
+
+	switch(ScrollInfo._sdir) {
+	case 0:
+		break;
+	case 2:
+		chunks++;
+	case 1:
+		sy -= 32;
+		x--;
+		y--;
+		blocks++;
+		break;
+	case 4:
+		blocks++;
+	case 3:

```

Comment:
SDIR_E and SDIR_SE swapped?
```suggestion
	case SDIR_E:
```